reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/http/RepoResource.scala

```diff
@@ -246,12 +252,20 @@ class RepoResource(keyserverClient: KeyserverClient, namespaceValidation: Namesp
           }
         } ~
         (pathEnd & put & entity(as[SignedPayload[TargetsRole]])) { signedPayload =>
+          // get them before they get removed from the DB
+          val previousTargetItemsF = targetItemRepo.findFor(repoId)
```

Comment:
this does not guarantee `findFor` is executed before `saveTargetRole`, this future is launched, but the future is never awaited on until after `saveTargetRole`. Depending on how the execution context schedules this future execution and the db pool, for example, this can easily have a different order.